File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/C/s910447421_commented.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    int x;
    int y;
} Ball;

int ans = 100;
int N;
Ball *balls;
int *taken;

int compare_x(const void *a, const void *b) {
    Ball *ballA = (Ball*)a;
    Ball *ballB = (Ball*)b;
    return (ballA->x - ballB->x);
}

void solve() {
    for (int add = 1; add < N; add++) {
        int p = balls[add].x - balls[0].x;
        int q = balls[add].y - balls[0].y;
        memset(taken, 0, N * sizeof(int));
        int current = check(p, q);
        if (current < ans) ans = current;
    }
}

int check(int p, int q) {
    int ret = 0;
    while (1) {
        int added = 0;
        int start = 0;
        int x, y;
        for (int i = 0; i < N; i++) {
            if (!taken[i]) {
                added = 1;
                ret++;
                taken[i] = 1;
                start = i;
                x = balls[i].x + p;
                y = balls[i].y + q;
                break;
            }
        }
        if (!added) break;
        for (int i = start + 1; i < N; i++) {
            if (balls[i].x == x && balls[i].y == y) {
                taken[i] = 1;
                x += p;
                y += q;
            }
        }
    }
    return ret;
}

int main() {
    scanf("%d", &N);
    ans = N;
    balls = malloc(N * sizeof(Ball));
    taken = malloc(N * sizeof(int));
    for (int i = 0; i < N; i++) {
        scanf("%d %d", &balls[i].x, &balls[i].y);
    }
    qsort(balls, N, sizeof(Ball), compare_x);
    solve();
    for (int i = 0; i < N; i++) {
        int temp = balls[i].x;
        balls[i].x = balls[i].y;
        balls[i].y = temp;
    }
    qsort(balls, N, sizeof(Ball), compare_x);
    solve();
    printf("%d\n", ans);
    free(balls);
    free(taken);
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/Go/s910447421_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

var ans int = 100
var balls [][]int
var N int

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)
	sc.Scan()
	N, _ = strconv.Atoi(sc.Text())
	ans = N
	balls = make([][]int, N)
	for i := 0; i < N; i++ {
		sc.Scan()
		x, _ := strconv.Atoi(sc.Text())
		sc.Scan()
		y, _ := strconv.Atoi(sc.Text())
		balls[i] = []int{x, y}
	}
	sort.Slice(balls, func(i, j int) bool {
		return balls[i][0] < balls[j][0]
	})
	solve()
	for i := 0; i < N; i++ {
		x := balls[i][0]
		y := balls[i][1]
		balls[i][0] = y
		balls[i][1] = x
	}
	sort.Slice(balls, func(i, j int) bool {
		return balls[i][0] < balls[j][0]
	})
	solve()
	fmt.Println(ans)
}

func solve() {
	for add := 1; add < N; add++ {
		p := balls[add][0] - balls[0][0]
		q := balls[add][1] - balls[0][1]
		taken := make([]bool, N)
		ret := check(p, q, taken)
		if ret < ans {
			ans = ret
		}
	}
}

func check(p, q int, taken []bool) int {
	ret := 0
	for {
		added := false
		var start int
		var x, y int
	Loop:
		for i := 0; i < N; i++ {
			if !taken[i] {
				added = true
				ret++
				taken[i] = true
				start = i
				x = balls[i][0] + p
				y = balls[i][1] + q
				break Loop
			}
		}
		if !added {
			break
		}
		for i := start + 1; i < N; i++ {
			if balls[i][0] == x && balls[i][1] == y {
				taken[i] = true
				x += p
				y += q
			}
		}
	}
	return ret
}

// End of Code
